File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/spring/viewcomponent/thymeleaf/application/web/index/IndexViewComponent.java
package de.tschuehly.spring.viewcomponent.thymeleaf.application.web.index;

import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;
import de.tschuehly.spring.viewcomponent.thymeleaf.EmptyViewContext;
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext;

@ViewComponent
public class IndexViewComponent {
    public ViewContext render(){
        return new EmptyViewContext();
    }
}

File: examples/jte-demo/src/main/kotlin/de/tschuehly/jteviewcomponentdemo/web/index/IndexViewComponent.kt
package de.tschuehly.jteviewcomponentdemo.web.index

import de.tschuehly.jteviewcomponentdemo.web.example.ExampleViewComponent
import de.tschuehly.jteviewcomponentdemo.web.para.ParameterViewComponent
import de.tschuehly.spring.viewcomponent.core.ViewComponent
import de.tschuehly.spring.viewcomponent.core.toProperty
import de.tschuehly.spring.viewcomponent.jte.ViewContext

@ViewComponent
class IndexViewComponent(
    private val exampleViewComponent: ExampleViewComponent,
    private val parameterViewComponent: ParameterViewComponent
) {

    fun render() = ViewContext(
        "exampleFormDTO" toProperty ExampleFormDTO("Spring ViewComponent"),
        "exampleViewComponent" toProperty exampleViewComponent,
        "parameterViewComponent" toProperty parameterViewComponent.render("Hello World")
    )

    class ExampleFormDTO(
        val exampleProp: String
    ) {
    }
}